README.md

```diff
@@ -1 +1,331 @@
+  ],
+  setupFilesAfterEnv: [
+    '<rootDir>/path/to/setup.js'
+  ]
+}
+```
+
+#### Using package.json
+
+```json
+{
+  "jest": {
+    "reporters": [
+      "default",
+      "<rootDir>/path/to/reporter.js"
+    ],
+    "setupFilesAfterEnv": [
+      "<rootDir>/path/to/setup.js"
+    ]
+  }
+}
+```
+
+### 2. Required Files
+
+Make sure both files are properly referenced in your Jest configuration:
+
+- **`setup.js`**: Must be included in `setupFilesAfterEnv` to intercept console calls
+- **`reporter.js`**: Must be included in `reporters` to generate the summary
+
+## How It Works
+
+The jest-console-count-reporter works by:
+
+1. **Setup Phase**: The `setup.js` file intercepts all console methods and tracks usage
+2. **Tracking**: Console calls are counted by method, message, and originating file
+3. **Aggregation**: Data is written to a temporary file (`/tmp/jest-console-counts.json`) 
+4. **Reporting**: The reporter reads the aggregate data and displays summaries
+
+**Important**: The setup file must be included in `setupFilesAfterEnv` for the reporter to function correctly.
+
+**Note**: The temporary aggregate file is automatically cleaned up at the start of each test run.
+
+## Usage
+
+Once configured, the reporter automatically tracks console usage during your test runs:
+
+```bash
+npm test
+```
+
+## Output Examples
+
+### Terminal Output
+
+The reporter provides a concise summary at the end of your test run:
+
+```
+[console-count] Console call summary:
+  log: 8
+  error: 5
+  warn: 4
+  debug: 2
+  info: 3
+  error: "Payment gateway timeout" - 1
+  warn: "Retrying payment" - 1
+
+Full summary available in /tmp/demo-project/coverage/console-count-summary.md
+```
+
+The terminal output shows:
+- **Method totals**: Count of each console method used across all tests
+- **Top messages**: The most recent or highest-count messages for some methods
+- **Markdown location**: Path to the detailed markdown report (when generated)
+
+### Markdown Report
+
+When your tests generate console output from 5 or more files, a detailed markdown report is created at `coverage/console-count-summary.md`:
+
+```markdown
+## [console-count] Console call summary:
+
+| Console Method | Total Count |
+|---|---|
+| `log` | 8 |
+| `error` | 5 |
+| `warn` | 4 |
+| `debug` | 2 |
+| `info` | 3 |
+
+## Top 5 `error` calls by message
+
+- (1) Payment gateway timeout
+  - (1) `demo-project/payment.test.js`
+- (1) Authentication failed
+  - (1) `demo-project/api.test.js`
+- (1) Validation error
+  - (1) `demo-project/api.test.js`
+
+## Top 5 `log` calls by message
+
+- (1) Processing payment
+  - (1) `demo-project/payment.test.js`
+- (1) Tracking user event
+  - (1) `demo-project/analytics.test.js`
+```
+
+The markdown report includes:
+- **Summary table**: Total counts per console method
+- **Top messages**: Up to 5 most frequent messages per method
+- **File breakdown**: Which files generated each message and how many times
+
+## Configuration Options
```

Comment:
Don't need the section altogether. It's not very much configurable - only having some variables on top. Don't mention configuration. If somebody is interested they will find their way through the vast amount of 4 files.